### ISSUE TITLE
Unhandled error in meteor install-sdk android

### ISSUE THREAD

**Issue by babnik63** [meteor/meteor#2978]
there was an error in my installation on ubuntu 14.4: I have jre- jdk and 32 bit utils for 64 linux

terminator@MyLinux:~$ meteor install-sdk android
Installing Android bundle
✓ A JDK is installed                                                            
✓ Found Android Platform tools                
Installing Android Build Tools                
                                              
events.js:72
        throw er; // Unhandled 'error' event
              ^
Error: read ECONNRESET
    at errnoException (net.js:904:11)
    at Pipe.onread (net.js:558:19)

**Comment by 0nRails** [comment]
Similar here
```
daniel@4TW:~/meteor_apps/microscope$ meteor install-sdk android
Installing Android bundle
Unexpected exit code 2 from bash [ '/home/daniel/.meteor/packages/meteor-tool/.1.0.38.1b3ywwh++os.linux.x86_64+web.browser+web.cordova/meteor-tool-os.linux.x86_64/tools/cordova-scripts/ensure_android_bundle.sh' ] 
stdout:
 
Installing Android development bundle.
This might take a while, please hold on.
Failed to install dependency kit.
 
stderr:
#########################################################                 79,4%
curl: (18) transfer closed with 90256775 bytes remaining to read

gzip: stdin: unexpected end of file
tar: Fin de archivo inesperado en el archivo
tar: Fin de archivo inesperado en el archivo
tar: Error is not recoverable: exiting now

Unexpected exit code from script: 2           
stdout:                                       
Installing Android development bundle.
This might take a while, please hold on.
Failed to install dependency kit.

#########################################################                 79,4%
curl: (18) transfer closed with 90256775 bytes remaining to read

gzip: stdin: unexpected end of file
tar: Fin de archivo inesperado en el archivo
tar: Fin de archivo inesperado en el archivo
tar: Error is not recoverable: exiting now

Failed to install android bundle              
```

**Comment by jrussell999** [comment]
I've got the exact same error as nicolsondsouza.
```
$ meteor install-sdk android
✓ Found Android SDK
✓ Found ant on PATH                           
✓ A JDK is installed                          
✓ Found Android Platform tools                
✓ Found Android Build Tools                   
✓ Found Android 19 API                        
✓ Found suitable Android x86 image            
Creating android virtual device (AVD): meteor 
                                              
events.js:72
        throw er; // Unhandled 'error' event
              ^
Error: read ECONNRESET
    at errnoException (net.js:904:11)
    at Pipe.onread (net.js:558:19)

$ meteor install-sdk android --verbose

which errors on:

android create avd --target 1 --name meteor --abi default/x86 --path /home/josh/.android/avd/meteor.avd
                                              
events.js:72
        throw er; // Unhandled 'error' event
              ^
Error: read ECONNRESET
    at errnoException (net.js:904:11)
    at Pipe.onread (net.js:558:19)
```
It always errors at the same point, and the connection did not break on my end.

**Comment by nicolsondsouza** [comment]
@jrussell999 

Hi, there I fixed this error in this manner,
go to $home/.meteor/anroid_bundle/
and remove ".android" folder
also remove "meteor_avd" folder,
and run 
meteor install-sdk android

This solved my issue, 
try this if you could fix the same.

**Comment by jrussell999** [comment]
@nicolsondsouza 
I decided to flatten & reinstall to fix this and a few other mistakes I made, but thanks anyway. My directory structure didn't look like that before, but it does now.

**Comment by myheren** [comment]
@babnik63 have you solve this problem yet, I've got a same one with you.

**Comment by zhonglijunyi** [comment]
@myheren have you solve this? I've got this too, i solved it as bellow:

First try: 
set "sdkman.force.http=true"(in .android/androidtool.cfg file). Or run >android, menu tools-> options, checked the check box [force https:// ...http://]. 
i thought it were ok, but it was not.

Then try: 
find command-cordova.js file under .meteor/packages/meteor-tool/..., then modify the line
    stdout = self.runAndroidTool(['update', 'sdk', '-s', '-t', target, '--all', '-u'], options);
    // added '-s' param, this means --no-https. 
then run > meteor install-sdk android, it was ok.

So we know, the commands-cordova.js does not handle the "androidtools.cfg" config file.

That worked for me. May it help you.

**Comment by babnik63** [comment]
I have the problem, but my pc is x86 and I think it has some drawbacks.


On Fri, Aug 7, 2015 at 9:45 PM, zhonglijunyi <notifications@github.com>
wrote:

> @myheren <https://github.com/myheren> have you solve this? I've got this
> too, i solved it as bellow:
>
> First try:
> set "sdkman.force.http=true"(in .android/androidtool.cfg file). Or run
> >android, menu tools-> options, checked the check box [force https://
> ...http://].
> i thought it were ok, but it was not.
>
> Then try:
> find command-cordova.js file under .meteor/packages/meteor-tool/..., then
> modify the line
> stdout = self.runAndroidTool(['update', 'sdk', '-s', '-t', target,
> '--all', '-u'], options);
> // added '-s' param, this means --no-https.
> then run > meteor install-sdk android, it was ok.
>
> So we know, the commands-cordova.js does not handle the "androidtools.cfg"
> config file.
>
> That worked for me. May it help you.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/meteor/meteor/issues/2978#issuecomment-128769878>.
>



-- 
Sincerely yours
Babak Niknia
Next Generation Solutions Co.
*Chief R&D Manager*
www.NEGSO.com

**Comment by EdwardOkech** [comment]
Hi guys am new to meteor, am trying to install it on my ubuntu 14.04lts machine and i keep getting this error
princek@princek-HP-ProBook-6540b:~$ curl https://install.meteor.com | sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  6121    0  6121    0     0    808      0 --:--:--  0:00:07 --:--:--  1630
Downloading Meteor distribution
################################################################          90.1%
curl: (18) transfer closed with 8307587 bytes remaining to read

gzip: stdin: unexpected end of file
tar: Unexpected EOF in archive
tar: Unexpected EOF in archive
tar: Error is not recoverable: exiting now
Installation failed.

any assistance will be highly appreciated.

**Comment by nicolsondsouza** [comment]
@EdwardOkech 

I see that you failed to get the bundle downloaded
in that case go here
https://warehouse.meteor.com/cordova/android_bundle_Linux_0.1.tar.gz
and download the bundle manually, from any of the download manager,
and then look at my instruction 
extract the bundle in ~/.meteor/android_bundle

Hi, there I fixed this error in this manner,
go to $home/.meteor/android_bundle/
and remove ".android" folder
also remove "meteor_avd" folder,
and run 
meteor install-sdk android

This solved my issue, 
try this if you could fix the same.

Or I can make a script, for that downloaded tar.gz, 
which will install it in your meteor directory,
but try this, and I was able to save a lot of bandwidth in my other pcs.
